scripts/postprocessing/coverage_analysis/coverage_Analyzer.py

```diff
@@ -0,0 +1,624 @@
+
+  #Returns the inverse (conjugate) of the quaternion given
+  def quat_inverse(self, robot_quat):
+      q_inv = [robot_quat[0], robot_quat[1], robot_quat[2], -robot_quat[3]]
+
+      return q_inv
+
+  def find_nearby_pose(self, testPt, fileIn, fileOut):
+      write_features = False
+
+      print("Finding nearby poses around: {:3f} {:3f} {:3f} {:3f} {:3f} {:3f} {:3f}".format(
+      testPt[1],testPt[2],testPt[3],testPt[4],testPt[5],testPt[6],testPt[7])) #skips ML num
+
+      with open(fileIn) as f:
+           f_out = open(fileOut, 'a')
+           lines = f.readlines()
+
+           for line in lines:
+               # Skip empty lines and those starting with comments
+               if re.match("^\s*\n", line) or re.match("^\s*\#", line):
+                  continue
+
+               # Extract all the values separated by spaces
+               vals = []
+               for v in re.split("\s+", line):
+                   if v == "":
+                      continue
+                   vals.append(v)
+
+               if (len(vals) == 8):
+                  ml_num = int(vals[0])
+                  robot_pose = [float(vals[1]), float(vals[2]), float(vals[3])]
+                  robot_quat = [float(vals[4]), float(vals[5]), float(vals[6]), float(vals[7])]
+
+                  test_pose = [testPt[1], testPt[2], testPt[3]]
+                  test_quat = [testPt[4], testPt[5], testPt[6], testPt[7]]
+
+                  #print("Checking pose {:d} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f}...".format(
+                  #ml_num,
+                  #robot_pose[0],robot_pose[1], robot_pose[2],
+                  #robot_quat[0], robot_quat[1], robot_quat[2], robot_quat[3]))
+
+                  #if (self.pose_error(robot_pose, test_pose) < self.max_pose_error and
+                  # self.quat_error(robot_quat, test_quat) < self.max_quat_error):
+
+                  #if (self.pose_error(robot_pose, test_pose) < self.max_pose_error):
+                  if (self.pose_inside_cube(robot_pose, test_pose, grid_size) == True):
+                     #print("Pose {:d} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f}... Passed".format(
+                     #     ml_num,
+                     #     robot_pose[0],robot_pose[1], robot_pose[2], 
+                     #     robot_quat[0], robot_quat[1], robot_quat[2], robot_quat[3]))
+
+                     #print("Passed")
+
+                     f_out.write('{:d} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f} {:.3f}\n'.format(
+                     ml_num,
+                     robot_pose[0], robot_pose[1], robot_pose[2],
+                     robot_quat[0], robot_quat[1], robot_quat[2], robot_quat[3]))
+
+                     write_features = True
+                     self.matchCounter += 1
+
+                  else:
+                     write_features = False
+                     ml_num = 0
+                     #print("Nope\n")
+
+
+               if (len(vals) == 3 and write_features and ml_num > 0):
```

Comment:
I don't believe this code will ever excute. write_features is true only if len(vals) is 8?